linera-storage/src/db_storage.rs

```diff
@@ -840,6 +880,36 @@ where
         Ok(exists)
     }
 
+    async fn list_events_from_index(
```

Comment:
Maybe `read_` instead of `list_`? (For `list` I would expect it to return only the indices, not the events, I think.)

Reply:
Ok.